docs/architecture/pairing-adapter-for-integration-system.md

```diff
@@ -0,0 +1,29 @@
+# Pairing Adapter for Integration System
+
+## Introduction
+
+When dealing with multiple Integration Systems each of them may support different ways of establishing trusted relation for the given Application instance. To provide a consistent experience for users, we introduce the concept of Pairing Adapters. Each Integration System can be coupled with one Pairing Adapter which will abstract the details on how the Integration System handles the trusted relation prerequisites.
+
+## Solution overview
+
+This diagram illustrates the flow for pairing the Application with its External Broker using the Pairing Adapter.
+
+![External Broker Pairing Adapter](./assets/external-broker-pairing-adapter.svg)
+
+>**NOTE** The Application must be created before the pairing is performed. To create an Application that is managed by the Integration System the Application should be created using Application Template.
+
+### Steps
+
+1. The Administrator requests the pairing data for the Application using the Cockpit (1).
+
+1. The Cockpit contacts the Director component to obtain the pairing data for the Application (2). The Application is loosely coupled with the Integration System that manages it. Thanks to this relation the Director component is aware which Pairing Adapter is assigned to the Integration System. Knowing this, the Pairing Adapter is called to request the pairing data (3).
+
+1. The Pairing Adapter calls External Broker to obtain pairing data (4).
+
+1. Administrator after obtaining the pairing data provides them to the Application instance which allows the Application instance to establish trusted relation with External Broker (6). This informs External Broker that there is a new Application instance that needs to be managed via this External Broker.
+
+## Details
+
+The Compass supports multiple different External Brokers (represented internally by Integration Systems) to help abstract the way how Applications and Runtimes can be managed. Each Integration System can have its own approach for obtaining details for establishing a trusted relation for the managed system. To simplify the process and improve users' experience we introduce the concept of Pairing Adapters. Each Integration System can be coupled with Pairing Adapter. Thanks to this approach the Directors' API can leverage different Integration Systems under single mutation and handle the requests in a consistent manner. When Administrator wants to establish a new trusted relation between Application or Runtime and Integration System the API offers `generateAuthForApplication(ID, type)` and `generateAuthForRuntime(ID, type)` mutations.
```

Comment:
- "supports multiple different" -> "supports multiple"
- "relation for the managed system" -> what managed system is... maybe: "relation with Application Instance"
- "To simplify the process..." -> "To make director architecture clean and improve users' experience ..."
- "...Runtime and Integration System the API offers.." -> "Runtime and Integration System the **Director** API offers.."
- about `generateAuthForApplication(ID, type)` and `generateAuthForRuntime(ID, type)` mutations. What with our existing ones, like genOneTimeToken(from connector) and genClientCreds. Should they be wrapped by those two above you mentioned or stay as separated ones?
  - Please add to the proposal information that IntegrationSystem pairing auth is only for registering app to compass, that means they will not be able to call Kyma Runtime APIs. Please think and write what they will need to do to get a correct credential to be able to call KymaRuntime APIs.

Reply:
Ad. 5. To offer consistent API we should migrate existing mutations to the new ones.